packages/cosmic-swingset/provisioning-server/src/ag_pserver/main.py

```diff
@@ -73,11 +73,27 @@ def errReceived(self, data):
     def processEnded(self, reason):
         self.deferred.callback((reason.value.exitCode, self.output, self.error))
 
+class StackedResource(resource.Resource):
+    def __init__(self, stack):
+        super().__init__()
+        self.stack = stack
+
+    def getChildWithDefault(self, *args):
+        res = super().getChildWithDefault(*args)
+        i = 0
```

Comment:
I don't think you need the up-call, your `StackedResource` has no content of it's own. Then the method might be clearer as: (warning, untested):

```python
def getChildWithDefault(self, *args):
    for s in self.stack:
        res = s.getChildWithDefault(*args)
        if not isinstance(res, resource.NoResource):
            return res
    return resource.NoResource("No such child resource.")
```